apache/netbeans
# Improve Jakarta EE 9/9.1 support

Changes for NetBeans in general

- Add schemas for Jakarta EE 9
- Add namespace for Jakarta EE
- Add missing deployment descriptor files for Java EE and Jakarta EE
- Add validations to create deployment descriptor for Jakarta EE 9
- Add mdd files for Jakarta EE 9 schemas
- Fix wrong version inside some mdd files
- Add support to create bean validation deployment descriptor's based on the Profile version
- Add DataLoader for Jakarta EE 9 schemas (EjbJar40DataLoader, DDWeb50DataLoader)
- Add missing DataLoader for Java EE 8 schemas (DDWeb40DataLoader)
- Add missing maven coordinates for Java/Jakarta EE
- Jakarta 9/9.1 xsd files have a license type of EDL-1.0 or EPL-v20
- Add some tests

Miscellaneous:
- Use try-with-resources
- Formatting code
- Add documentation
- Use diamond operator
- Use StandardCharsets
- Remove unused imports
- Use switch-case instead of long if-else if-else
- Use Multi-catch when possible
- Improve performance of Collection.toArray

Testing:
- Full build done
- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Sigtests
- Verify successful execution of commit-validation
- Verify successful execution of rat-exclusions
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Download every version of GlassFish and successfully:
  - Create Web App Project from Java EE 5 to Jakarta EE 9.1
  - Create every deployment descriptor from Java EE 5 to Jakarta EE 9.1 in each project
  - Verify correct version of each deployment descriptor match to the Profile version of the Web App
  - Deploy and run the project
